stress: failed test in cockroach/kv/kv.test: TestTxnDBLostUpdateAnomaly_Delete

Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/fbb1243f55df4aa02f59c57d4133ec85460748e7

Stress build found a failed test:

```
=== RUN   TestTxnDBLostUpdateAnomaly_Delete
W160425 04:26:13.732163 gossip/gossip.go:892  not connected to cluster; use --join to specify a connected node
I160425 04:26:13.732194 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
I160425 04:26:13.733389 kv/txn_correctness_test.go:517  verifying all possible histories for the "lost update (delete)" anomaly
I160425 04:26:13.734332 storage/replica_command.go:1439  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 1s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=1970-01-01 00:00:00 +0000 UTC]
E160425 04:26:13.734543 config/config.go:382  unable to determine largest reserved object ID from system config: descriptor table not found in system config of 0 values
--- FAIL: TestTxnDBLostUpdateAnomaly_Delete (3.82s)
	txn_correctness_test.go:572: ([I1(A) C1 D3(A) D3(B) R2(A) C3 W2(B-A) C2]): unexpected failure running D3(A): txn "kv/txn_correctness_test.go:622 txn2" id=ea58e08c key="1699.B" rw=false pri=0.00000014 iso=SERIALIZABLE stat=PENDING epo=0 ts=0.000000000,7191 orig=0.000000000,7191 max=0.000000000,7191 wto=false: non-aborted transaction exists already
```

Run Details:

```
0 runs so far, 0 failures, over 5s
0 runs so far, 0 failures, over 10s
0 runs so far, 0 failures, over 15s
0 runs so far, 0 failures, over 20s
5 runs so far, 0 failures, over 25s
11 runs so far, 0 failures, over 30s
14 runs so far, 0 failures, over 35s
16 runs so far, 0 failures, over 40s
16 runs so far, 0 failures, over 45s
18 runs so far, 0 failures, over 50s
24 runs so far, 0 failures, over 55s
29 runs so far, 0 failures, over 1m0s
31 runs so far, 0 failures, over 1m5s
32 runs so far, 0 failures, over 1m10s
35 runs so far, 0 failures, over 1m15s
36 runs so far, 0 failures, over 1m20s
43 runs so far, 0 failures, over 1m25s
47 runs so far, 0 failures, over 1m30s
48 runs so far, 0 failures, over 1m35s
48 runs so far, 0 failures, over 1m40s
51 runs so far, 0 failures, over 1m45s
54 runs so far, 0 failures, over 1m50s
59 runs so far, 0 failures, over 1m55s
64 runs so far, 0 failures, over 2m0s
64 runs so far, 0 failures, over 2m5s
65 runs so far, 0 failures, over 2m10s
69 runs so far, 0 failures, over 2m15s
75 runs so far, 0 failures, over 2m20s
77 runs so far, 0 failures, over 2m25s
81 runs so far, 0 failures, over 2m30s
81 runs so far, 0 failures, over 2m35s
84 runs so far, 0 failures, over 2m40s
87 runs so far, 0 failures, over 2m45s
90 runs so far, 0 failures, over 2m50s
94 runs so far, 0 failures, over 2m55s
97 runs so far, 0 failures, over 3m0s
99 runs so far, 0 failures, over 3m5s
101 runs so far, 0 failures, over 3m10s
104 runs so far, 0 failures, over 3m15s
108 runs so far, 0 failures, over 3m20s
111 runs so far, 0 failures, over 3m25s
113 runs so far, 0 failures, over 3m30s
116 runs so far, 0 failures, over 3m35s
119 runs so far, 0 failures, over 3m40s
123 runs so far, 0 failures, over 3m45s
127 runs so far, 0 failures, over 3m50s
129 runs so far, 0 failures, over 3m55s
132 runs so far, 0 failures, over 4m0s
135 runs so far, 0 failures, over 4m5s
138 runs so far, 0 failures, over 4m10s
143 runs so far, 0 failures, over 4m15s
144 runs so far, 0 failures, over 4m20s
147 runs so far, 0 failures, over 4m25s
148 runs so far, 0 failures, over 4m30s
151 runs so far, 0 failures, over 4m35s
155 runs so far, 0 failures, over 4m40s
159 runs so far, 0 failures, over 4m45s
162 runs so far, 0 failures, over 4m50s
163 runs so far, 0 failures, over 4m55s
166 runs so far, 0 failures, over 5m0s
168 runs so far, 0 failures, over 5m5s
173 runs so far, 0 failures, over 5m10s
176 runs so far, 0 failures, over 5m15s
177 runs so far, 0 failures, over 5m20s
179 runs so far, 0 failures, over 5m25s
181 runs so far, 0 failures, over 5m30s
185 runs so far, 0 failures, over 5m35s
189 runs so far, 0 failures, over 5m40s
193 runs so far, 0 failures, over 5m45s
193 runs so far, 0 failures, over 5m50s
196 runs so far, 0 failures, over 5m55s
197 runs so far, 0 failures, over 6m0s
202 runs so far, 0 failures, over 6m5s
207 runs so far, 0 failures, over 6m10s
207 runs so far, 0 failures, over 6m15s
209 runs so far, 0 failures, over 6m20s
212 runs so far, 0 failures, over 6m25s
215 runs so far, 0 failures, over 6m30s
219 runs so far, 0 failures, over 6m35s
220 runs so far, 0 failures, over 6m40s
222 runs so far, 0 failures, over 6m45s
224 runs so far, 0 failures, over 6m50s
228 runs so far, 0 failures, over 6m55s
232 runs so far, 0 failures, over 7m0s
234 runs so far, 0 failures, over 7m5s
235 runs so far, 0 failures, over 7m10s
238 runs so far, 0 failures, over 7m15s
241 runs so far, 0 failures, over 7m20s
245 runs so far, 0 failures, over 7m25s
248 runs so far, 0 failures, over 7m30s
250 runs so far, 0 failures, over 7m35s
253 runs so far, 0 failures, over 7m40s
255 runs so far, 0 failures, over 7m45s
258 runs so far, 0 failures, over 7m50s
262 runs so far, 0 failures, over 7m55s
264 runs so far, 0 failures, over 8m0s
266 runs so far, 0 failures, over 8m5s
270 runs so far, 0 failures, over 8m10s
271 runs so far, 0 failures, over 8m15s
274 runs so far, 0 failures, over 8m20s
278 runs so far, 0 failures, over 8m25s
281 runs so far, 0 failures, over 8m30s
282 runs so far, 0 failures, over 8m35s
286 runs so far, 0 failures, over 8m40s
290 runs so far, 0 failures, over 8m45s
291 runs so far, 0 failures, over 8m50s
296 runs so far, 0 failures, over 8m55s
297 runs so far, 0 failures, over 9m0s
300 runs so far, 0 failures, over 9m5s
305 runs so far, 0 failures, over 9m10s
307 runs so far, 0 failures, over 9m15s
310 runs so far, 0 failures, over 9m20s
312 runs completed, 1 failures, over 9m21s
FAIL
```
Please assign, take a look and update the issue accordingly.

--- a/storage/replica_command.go
+++ b/storage/replica_command.go
@@ -351,7 +351,7 @@ func (r *Replica) BeginTransaction(
 			// command's txn and rewrite the record.
 			reply.Txn.Update(&txn)
 		} else {
-			return reply, roachpb.NewTransactionStatusError("non-aborted transaction exists already")
+			return reply, roachpb.NewTransactionStatusError(fmt.Sprintf("BeginTransaction can't overwrite %s", txn))
 		}
 	}
 

